src/Resources/app/storefront/src/checkout/checkout.plugin.js

```diff
@@ -35,14 +35,26 @@ export default class CheckoutPlugin extends Plugin {
         const confirmPaymentForm = DomAccess.querySelector(document, '#confirmPaymentForm');
         confirmPaymentForm.addEventListener('submit', this.onConfirmPayment.bind(this));
 
-        const formattedHandlerIdentifier = 'handler_adyen_cardspaymentmethodhandler';
+        const cardsFormattedHandlerIdentifier = 'handler_adyen_cardspaymentmethodhandler';
+        const idealFormattedHandlerIdentifier = 'handler_adyen_idealpaymentmethodhandler';
+        const klarnaPayNowFormattedHandlerIdentifier = 'handler_adyen_klarnapaynowpaymentmethodhandler';
+        const klarnaPayLaterFormattedHandlerIdentifier = 'handler_adyen_klarnapaylaterpaynowpaymentmethodhandler';
+        const sepaFormattedHandlerIdentifier = 'handler_adyen_sepapaymentmethodhandler';
+        const sofortFormattedHandlerIdentifier = 'handler_adyen_sofortpaymentmethodhandler';
+        const paypalFormattedHandlerIdentifier = 'handler_adyen_paypalpaymentmethodhandler';
 
         this.paymentMethodTypeHandlers = {
-            'scheme': formattedHandlerIdentifier
+            'scheme': cardsFormattedHandlerIdentifier,
+            'ideal': idealFormattedHandlerIdentifier,
+            'klarna': klarnaPayLaterFormattedHandlerIdentifier,
+            'klarna_paynow': klarnaPayNowFormattedHandlerIdentifier,
+            'sepadirectdebit': sepaFormattedHandlerIdentifier,
+            'sofort': sofortFormattedHandlerIdentifier,
+            'paypal': paypalFormattedHandlerIdentifier
         };
 
         //PMs that should show an 'Update Details' button if there's already a state.data for that PM stored for this context
-        this.updatablePaymentMethods = ['scheme'];
+        this.updatablePaymentMethods = ['scheme', 'ideal', 'sepadirectdebit'];
```

Comment:
let's create a ticket to see if we can assume that a payment method is update-able if it has payment details. In that case we won't need to manually update this list.
We only need payment methods in this list when they have some fields rendered by the checkout component which is indicated by the payment.details object so I think it should work.

Reply:
PW-3122